internal/pkg/template/api/extension/token/token.go

```diff
@@ -0,0 +1,268 @@
+}
+
+const (
+	MetaKeySchemeName  = "keboola:scheme-name"
+	MetaKeyTokenHeader = "keboola:token-header" // nolint: gosec
+)
+
+// AddTokenHeaderToPayloads adds token header to each endpoint with the tokenScheme.
+// Token "header" value will be accessible in the payload as the "field".
+//
+// AddTokenHeaderToPayloads must appear in a Service expression.
+//
+// Examples:
+//
+//    var tokenSecurity = APIKeySecurity("storage-api-token", func() {
+//	     Description("Storage Api Token Authentication")
+//    })
+//
+//    var _ = Service("templates", func() {
+//	    Description("Service for applying templates to Keboola projects")
+//
+//	    Security(tokenSecurity)
+//	    defer AddTokenHeaderToPayloads(tokenSecurity, "storageApiToken", "X-StorageApi-Token")
+//
+//      Method("with-no-security", func() {
+//        NoSecurity()
+//        HTTP(func() {
+//          GET("/no-security")
+//		  })
+//	    })
+//
+//      Method("with-token-security", func() {
+//        HTTP(func() {
+//          GET("/token-security")
+//		  })
+//	    })
+//    }
+//
+func AddTokenHeaderToPayloads(tokenScheme *expr.SchemeExpr, field, header string) {
+	service, ok := eval.Current().(*expr.ServiceExpr)
+	if !ok {
+		eval.IncompatibleDSL()
+		return
+	}
+
+	// Add token header to the service metadata
+	eval.Execute(func() {
+		Meta(MetaKeySchemeName, tokenScheme.SchemeName)
+		Meta(MetaKeyTokenHeader, header)
+	}, service)
+
+	// Iterate over methods
+	for _, m := range service.Methods {
+		// Modify method payload
+		method := m
+		methodFn := method.DSLFunc
+		method.DSLFunc = func() {
+			// Invoke original definitions
+			methodFn()
+
+			// Use default security from the Service, if no security is set
+			requirements := method.Requirements
+			if len(requirements) == 0 {
+				requirements = service.Requirements
+			}
+
+			// Iterate over all security schemes
+			for _, requirement := range requirements {
+				for _, scheme := range requirement.Schemes {
+					if scheme.SchemeName == tokenScheme.SchemeName {
+						// Add token to the payload definition
+						if method.Payload == nil {
+							// Init payload
+							Payload(func() {
+								APIKey(scheme.SchemeName, field, String)
+								Required(field)
+							})
+						} else {
+							// Add ti payload
+							eval.Execute(func() {
+								APIKey(scheme.SchemeName, field, String)
+								Required(field)
+							}, method.Payload)
+						}
+
+						// Add header to the HTTP definition
+						endpoint := expr.Root.API.HTTP.ServiceFor(method.Service).EndpointFor(method.Name, method)
+						httpFn := endpoint.DSLFunc
+						endpoint.DSLFunc = func() {
+							// Define the payload field by the header
+							Header(field + ":" + header)
+
+							// Invoke original definitions
+							if httpFn != nil {
+								httpFn()
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+}
+
+func Generate(_ string, roots []eval.Root, files []*codegen.File) ([]*codegen.File, error) {
```

Comment:
Goa je pre nas prilis univerzalna a ukecana:
- generuje do kazdeho endpointu `Header` ohladom autorizacie, co nie je potrebne (staci link na security).
- generuje vlastny nazov `security`, vid zelena sipka, to tam nechceme.
- po uprave suborov dostaneme pekny vysledok, ktory sa nelisi od toho, ako by sa to napisalo rucne.

![image](https://user-images.githubusercontent.com/19371734/158208955-0bb23774-8fb3-4bbb-9638-620acd9efa4c.png)